bin/admin/clang-format.sh

```diff
@@ -0,0 +1,92 @@
+#!/bin/sh
+
+# these are the versions of clang-format that are supported required
+supported_clang_format_versions="16 17"
+preferred_clang_format_version=""  # prefer most recent supported clang-format version
+for v in $supported_clang_format_versions; do
+  preferred_clang_format_version=$v
+done
+
+# append common locations of clang-format to PATH
+unameOut="$(uname -s)"
+case "${unameOut}" in
+    Darwin*)
+      extra_path=""
+      for v in $supported_clang_format_versions; do
+        extra_path=/opt/homebrew/opt/llvm@$v/bin:/opt/homebrew/opt/clang-format@$v/bin:$extra_path
+      done
+      # postpend paths, but preferring more recent versions
+      export PATH=$PATH:$extra_path:/opt/homebrew/bin
+    ;;
+esac
+
+path_to_clang_format=`which clang-format`
+have_supported_clang_format_version=0
+if [[ "X$path_to_clang_format" != "X" ]]; then
```

Comment:
This syntax, `[[` and `]]`, is a bash-extension and thus the script does not work on Debian/Ubuntu, where `/bin/sh` is symlinked to `dash`.

A solution is to change the shebang to `#!/bin/bash`, or use `[` and `]`.